heartbeat/ZFS

```diff
@@ -0,0 +1,198 @@
+#!/bin/sh
+#
+# License:      GNU General Public License (GPL)
+# Support:      zfs@lists.illumos.org
+# Written by:   Saso Kiselkov
+#
+#	This script manages ZFS pools
+#	It can import a ZFS pool or export it
+#
+#	usage: $0 {start|stop|status|monitor|validate-all|meta-data}
+#
+#	The "start" arg imports a ZFS pool.
+#	The "stop" arg exports it.
+#
+#       OCF parameters are as follows
+#       OCF_RESKEY_pool - the pool to import/export
+#
+#######################################################################
+# Initialization:
+
+: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
+. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
+
+# Set the variable below to '1' to enable zpool import/export logging.
+# This is useful to determine how long these operations take on your system.
+# DEBUG=0
+DEBUGLOG="/var/log/ZFS_cluster_debug.log"
+
+USAGE="usage: $0 {start|stop|status|monitor|validate-all|meta-data}";
+
+#######################################################################
+
+debug_log () {
+    if [ "$DEBUG" -eq 1 ]; then
+        echo "$(date):${OCF_RESKEY_pool}:" "$@" >> "$DEBUGLOG"
+    fi
+}
+
+meta_data() {
+        cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="ZFS">
+<version>1.0</version>
+<longdesc lang="en">
+This script manages ZFS pools
+It can import a ZFS pool or export it
+</longdesc>
+<shortdesc lang="en">Manages ZFS pools</shortdesc>
+
+<parameters>
+<parameter name="pool" unique="1" required="1">
+<longdesc lang="en">
+The name of the ZFS pool to manage, e.g. "tank".
+</longdesc>
+<shortdesc lang="en">ZFS pool name</shortdesc>
+<content type="string" default="" />
+</parameter>
+<parameter name="importargs" unique="0" required="0">
+<longdesc lang="en">
+Arguments to zpool import, e.g. "-d /dev/disk/by-id".
+</longdesc>
+<shortdesc lang="en">Import arguments</shortdesc>
+<content type="string" default="" />
+</parameter>
+<parameter name="importforce" unique="1" required="0">
+<longdesc lang="en">
+zpool import is given the -f option.
+</longdesc>
+<shortdesc lang="en">Import is forced</shortdesc>
+<content type="boolean" default="true" />
```

Comment:
You should define and assign "_default" parameters to use here and assign if not set by user (grep for _default in the other agents for examples).